crates/ethportal-api/src/types/execution/receipts.rs

```diff
@@ -1,56 +1,33 @@
-use std::{
-    ops::{Deref, DerefMut},
-    sync::Arc,
-};
-
 use alloy::{
-    primitives::{Address, Bloom, BloomInput, B256, U256},
-    rlp::{
-        length_of_length, Decodable, Encodable, Error as RlpError, Header as RlpHeader,
-        RlpDecodable, RlpEncodable,
-    },
+    consensus::{proofs::calculate_receipt_root, ReceiptEnvelope},
+    eips::{Decodable2718, Encodable2718},
+    primitives::B256,
+    rlp::{Decodable, Encodable, Header as RlpHeader},
 };
-use anyhow::anyhow;
-use bytes::{Buf, BufMut, Bytes};
-use eth_trie::{EthTrie, MemoryDB, Trie};
+use bytes::BufMut;
 use serde::{Deserialize, Deserializer};
 use serde_json::Value;
 
-use super::transaction::JsonBytes;
-use crate::utils::bytes::hex_decode;
-
 // 2 ^ 14
 const MAX_TRANSACTION_COUNT: usize = 16384;
 
 /// Represents the `Receipts` datatype used by the chain history wire protocol
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Receipts {
```

Comment:
Any reason we're not using https://docs.rs/alloy/latest/alloy/consensus/struct.Receipts.html ?

Reply:
Alloy Receipts is a `A two-dimensional vector of [Receipt](https://docs.rs/alloy/latest/alloy/consensus/struct.Receipt.html) instances` where as we are currently using a one dimensional vector of receipts. I don't think using `Receipts` would really add much benefit other then potentially break things, so I don't feel it is worth the effort to possibly use it, as realistically it doesn't change anything it would if anything make the code less clean, as we would still need to write our code to encode the receipts as a 1 dimensional array I am assuming.

Maybe somebody could make a case for it in the future?, but currently I can't justify reading into it further.